### PR TITLE
CE-172 ReactOnRails v9 + Webpacker v3

### PR DESCRIPTION
**Pull request by toastercup**
## `cortex-plugins-core` Purpose:
* `cortex-plugins-core` is now truly packaged for consumption - previously, we had hacks in place that would require `cortex` perform the babel transpilation. Now, we simply package up the transpiled sources in our Yarn/Npm package. This means we can use the default Webpack/Webpacker configs, resulting in easier Rails upgrades in the future.
* Package only includes bare minimum set of files via use of `.npmignore` and `.gitignore`
* Implements transpilation/build Yarn tasks. These will be included in `cortex-plugins-core`'s documentation once more of the library's infrastructure is finalized.
* Removed `react-on-rails` dependency - `plugins-core` doesn't need this, as it doesn't mount its own components directly in the DOM.

## `cortex` Purpose:
* Migrates to Rails 5.1's Webpacker/React infrastructure, relegating ReactOnRails to a much simpler role. Upgrades ReactOnRails to v9. This new setup means we (hopefully) shouldn't have to make any major upgrades to our React infrastructure in the near future.
* Hot/live reloading for local development now fully functional!
* Removes messy, no-longer-necessary plugin/asset/testing logic for Webpack
* Prerender enabled by default for every environment except `development`
* Miscellaneous refactors

## JIRA:
https://careerbuilder.atlassian.net/browse/CE-172

## Steps to Take On Prod
N/A

## Changes:
* Changes to setup
  * N/A

* Architectural changes
  * N/A

* Migrations
  * N/A

* Library changes
  * `cortex-plugins-core` requires a bump

* Side effects
  * Let's be sure to test prerender functionality on Staging/Production

## Screenshots
* Before
N/A

* After
N/A

## QA Links:
http://web.cortex-2.development.c66.me/

## How to Verify These Changes
* Specific pages to visit
  * N/A

* Steps to take
  * Ensure Tenant Switcher functions normally
  * Ensure beta Media Create page functions normally

* Responsive considerations
  * N/A

## Relevant PRs/Dependencies:
https://github.com/cbdr/cortex/pull/531

## Additional Information
N/A

### CODE REVIEW

File: cortex-plugins-core.gemspec [hunk 1]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |s|
   s.files = Dir["{app,config,db,lib}/**/*", "LICENSE.md", "Rakefile", "README.md"]
 
   s.add_dependency "rails", ">= 5"
-  s.add_dependency "react_on_rails", "~> 8"
+  s.add_dependency "react_on_rails", "~> 9"
```
Comment (MKwenhua) [inline cortex-plugins-core.gemspec]: I don't think this dependency will be necessary since these components wont be mounting on their own.

**Review by MKwenhua — CHANGES_REQUESTED** [review]

File: cortex-plugins-core.gemspec (hunk 1 above)
Comment (toastercup) [reply cortex-plugins-core.gemspec]: Good catch! I've removed this superfluous dependency from the `gemspec` and `package.json`

**Comment by toastercup** [comment]
@MKwenhua ready for re-review

**Review by MKwenhua — APPROVED** [review]

**Comment by MKwenhua** [comment]
@toastercup Ok approved.

**Review by mrfantasticwonders — APPROVED** [review]

**Pull request merged**